source/renderer/app/containers/wallet/WalletUtxoPage.js

```diff
@@ -39,6 +39,9 @@ export default class WalletSettingsPage extends Component<Props> {
 
     return (
       <WalletUtxo
+        isLoadingInitialUtxoData={
+          walletSettings.getWalletUtxosRequest.isExecutingFirstTime
```

Comment:
@DominikGuzei shouldn't we make sure we check if this request has been executed for the first time **for the active wallet**? We fetch UTXO distribution data for each wallet and I am afraid we will have this working only for the first one (or if the user has only one wallet)...
If I remember correctly, it should be possible to pass on walletId to check if the request has been ever executed for the given wallet?

Reply:
Great catch 👍

Reply:
@nikolaglumac my latest commit fixes this by resetting the utxo request when leaving the utxo page. This also means that it will show the spinner also anytime you go back to the utxo page within the same wallet. But it's the easiest solution and I think ok since the spinner is rarely showing up anyway - what do you think?